doc/ipython-notebooks/neuralnets/autoencoders.ipynb

```diff
@@ -391,7 +399,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "conv_nn = ae.convert_to_neural_network(NeuralSoftmaxLayer(10))\n",
+    "conv_nn = sg.convert_to_neural_network(ae, sg.layer(\"NeuralSoftmaxLayer\", num_neurons=10), 0.01)\n",
```

Comment:
@karlnapf @vigsterkr would this be a reasonable tradeoff? Imo much more readable than putting an output layer and then calling get() to get a net Machine out...

Reply:
also, each free function adds about 100 LoC in SWIG

Reply:
I say do it.

But one question: Could we maybe overload the factory for this?

Reply:
Hmmm not sure I understand. You mean the cast functions?

Reply:
the factory for labels can for example take a vector or a csv file and then returns a labels object, potentially converted. See what I mean?

Reply:
hmm, I see. You mean `sg.machine(ae, sg.layer("NeuralSoftmaxLayer", num_neurons=10), sigma=0.01)` would convert DeepAutoencoder to a NeuralNetwork with an output layer for inference?

Reply:
Something in those lines

Reply:
OK. I can look into this in another PR

Reply:
actually rather than using the factory, we should use the casting functions we have `as_neural_network`, `as_kernel`, etc

Reply:
OK, I'll write an issue for this then!